src/act.informative.cpp

```diff
@@ -3843,6 +3843,10 @@ ACMD(do_scan) {
     int from_room, to_room;
     Exit *exit;
 
+    if (GET_CLASS(ch) != CLASS_ROGUE) {
+        GET_HIDDENNESS(ch) = 0;
+    }
+
```

Comment:
You should hide the "$N scans the area" room act below if they're hidden.

Reply:
It is hidden.  I forgot to actually stop hiddenness from being removed